### DISCUSSION TITLE
Prisma Invalid created_at ISO-8601 DateTime Format Error in Query

### DISCUSSION THREAD

**Discussion by Krrunal** [prisma/prisma#25295] [Q&A]
### Question

When trying to perform a query using Prisma to filter records based on a date range, I encountered the following error when passing ISO-8601 formatted date strings.

### Information about Prisma Schema, Client Queries and Environment (optional)

**Code Snippet:**
```
prisma.package.findMany({
  where: {
    created_at: {
      gte: "2024-09-12", // ISO-8601 formatted string
      lte: "2024-09-13"
    },
    voys_status: {
      not: ""
    }
  },
  include: {
    store: true,
    route: true
  }
});
```
**Error: **
Invalid value for argument `gte`: premature end of input. Expected ISO-8601 DateTime.

**Full error message:**
PrismaClientValidationError: 
Invalid `prisma.package.findMany()` invocation:

{
  where: {
    created_at: {
      gte: "2024-09-12",
           ~~~~~~~~~~~~
      lte: "2024-09-13"
    },
    voys_status: {
      not: ""
    }
  },
  include: {
    store: true,
    route: true
  }
}

Invalid value for argument `gte`: premature end of input. Expected ISO-8601 DateTime.

**Context**
Context:
I am trying to filter records from the packages database table where the created_at field falls between two specific dates, specifically 2024-09-12 and 2024-09-13. The intent is to fetch all records for the specified date without considering the time component (00:00:00 to 23:59:59).

**Problem:**
When I use the following code to filter the records:


```
created_at: {
  gte: "2024-09-12",
  lte: "2024-09-13"
}
```
**I receive the following error:**


Invalid value for argument `gte`: premature end of input. Expected ISO-8601 DateTime.
This indicates that the date format used does not match the expected format in Prisma.

**Attempted Solutions:**
When I try to adjust my filtering criteria using JavaScript Date objects:


```
created_at: {
  lte: new Date("2024-09-13"),
  gte: new Date("2024-09-12")
}
```
This returns a blank object, indicating no matching records were found, despite there being records in the database that meet the criteria.

**Expected Behavior:**
I would expect the query to return all records from the packages table for the specified date range, treating the dates as full days without time constraints.

**Current Database Format:**
The created_at field in the database is stored in the format YYYY-MM-DD HH:MM:SS.SSS, which means I need to find a way to effectively compare this format with the provided date values.

**Comment by ludralph** [comment]
Hi @Krrunal 👋 

Can you please check this similar [discussion](https://github.com/prisma/prisma/discussions/20219#discussioncomment-6448028) that addresses how you can filter by date without the time?

**Reply by DrJume** [reply]
We have the same issue. We are saving ISO Dates as regular Strings. Using `lte`/`gte` with String fields should only do string comparison and shouldn't require to transform the findMany() args with `new Date()`.

**Reply by DrJume** [reply]
But passing the findMany arg with new Date() gives the following error:
```
Argument gte: Invalid value provided. Expected String or StringFieldRefInput, provided DateTime.
```

**Reply by DrJume** [reply]
It turned out to be a bug with the error logging. Because of a Promise.all() multiple prisma client requests were dispatched simultaneously, leading one failure to display the wrong source code context/stacktrace (wrongly attributing the first prisma call in Promise.all).
The error was passing an YYYY-MM-DD string to a normal DateTime column.

**Comment by ojtorres-mx** [comment]
dont use prisma, is a pain in the neck for working with dates
